pkg/utils/secrets/basic_auth.go

```diff
@@ -93,26 +111,52 @@ func (b *BasicAuth) SecretData() map[string][]byte {
 	case BasicAuthFormatNormal:
 		data[DataKeyUserName] = []byte(b.Username)
 		data[DataKeyPassword] = []byte(b.Password)
+
+		if b.BcryptPasswordHash != "" {
+			data[DataKeyPasswordHash] = []byte(b.BcryptPasswordHash)
+		}
+
 		fallthrough
 
 	case BasicAuthFormatCSV:
-		data[DataKeyCSV] = []byte(fmt.Sprintf("%s,%s,%s,%s", b.Password, b.Username, b.Username, user.SystemPrivilegedGroup))
+		var csvFormatStr bytes.Buffer
+
+		if b.BcryptPasswordHash != "" {
+			csvFormatStr.WriteString(b.BcryptPasswordHash + ",")
+		}
+
+		csvFormatStr.WriteString(fmt.Sprintf("%s,%s,%s,%s", b.Password, b.Username, b.Username, user.SystemPrivilegedGroup))
+		data[DataKeyCSV] = []byte(csvFormatStr.String())
 	}
 
 	return data
 }
 
-// LoadBasicAuthFromCSV loads the basic auth. username and the password from the given CSV-formatted <data>.
+// LoadBasicAuthFromCSV loads the basic auth. username, password and the password hash from the given CSV-formatted <data>.
 func LoadBasicAuthFromCSV(name string, data []byte) (*BasicAuth, error) {
 	csv := strings.Split(string(data), ",")
 	if len(csv) < 2 {
 		return nil, fmt.Errorf("invalid CSV for loading basic auth data: %s", string(data))
 	}
 
-	return &BasicAuth{
-		Name: name,
+	var basicAuth *BasicAuth
```

Comment:
With respect to what I write above, I think that following lines change should be reverted.